### ISSUE TITLE
mender_grub_storage_device not exported in grub.d mode

### ISSUE THREAD

**Issue by chgabriel79** [mendersoftware/grub-mender-grubenv#33]
In grub.d mode (when integrated into the /etc/grub.d framework), the variable `mender_grub_storage_device` is set in the `grub.cfg` from the boot partition but is also used in the config from the root partition. Since it is never set there, the boot fails.

The following simple patch resolves this problem:

```diff
--- a/grub.d/00_05_mender_setup_env_grub
+++ b/grub.d/00_05_mender_setup_env_grub
@@ -6,5 +6,6 @@ if [ "$GRUB_MENDER_GRUBENV_CFG_GENERATION" != "true" ]; then
 mender_check_and_restore_env
 mender_load_env_with_rollback
 regexp (.*),(.*) $root -s mender_grub_storage_device
+export mender_grub_storage_device
 EOF
 fi
```

Please consider applying.

**Comment by kacf** [comment]
Thanks @chgabriel79, I believe this is a good fix. I will test it.

FYI, we are not actually using Github issues, but [Jira](https://northerntech.atlassian.net/jira/software/c/projects/MEN/issues) tickets. Issues were left enabled in this repository by mistake. So I will close down the Issues tab after this is handled.

**Comment by kacf** [comment]
https://github.com/mendersoftware/mender-convert/pull/550

**Comment by kacf** [comment]
Merged, and cherry-picked to 3.0.x and 4.0.x.

Now closing Issues in this repository.
